src/java/htsjdk/samtools/SAMRecord.java

```diff
@@ -1394,42 +1394,22 @@ protected void setAttribute(final short tag, final Object value) {
      * @param value the value to be checked
      * @return true if the value is valid and false otherwise
      */
+    @Deprecated
```

Comment:
I actually made it protected in SAMBinaryTagAndValue on the theory that it doesn't need to be public since its enforced by SAMBinaryTagAndValue now. But let me know if you think it needs to be public there too.